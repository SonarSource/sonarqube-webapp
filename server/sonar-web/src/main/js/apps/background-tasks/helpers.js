/*
 * SonarQube
 * Copyright (C) 2009-2016 SonarSource SA
 * mailto:contact AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
const ONE_SECOND = 1000;
const ONE_MINUTE = 60 * ONE_SECOND;

function format (int, suffix) {
  return `${int}${suffix}`;
}

export function formatDuration (value) {
  if (!value) {
    return '';
  }
  if (value >= ONE_MINUTE) {
    const minutes = Math.round(value / ONE_MINUTE);
    return format(minutes, 'min');
  } else if (value >= ONE_SECOND) {
    const seconds = Math.round(value / ONE_SECOND);
    return format(seconds, 's');
  } else {
    return format(value, 'ms');
  }
}
