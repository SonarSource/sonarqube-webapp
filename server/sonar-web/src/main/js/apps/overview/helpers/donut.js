import d3 from 'd3';
import React from 'react';

let Sector = React.createClass({
  render() {
    let arc = d3.svg.arc()
        .outerRadius(this.props.radius)
        .innerRadius(this.props.radius - this.props.thickness);
    return <path d={arc(this.props.data)} style={{ fill: this.props.fill }}/>;
  }
});

export default React.createClass({
  getDefaultProps() {
    return {
      size: 30,
      thickness: 6
    };
  },

  render() {
    let radius = this.props.size / 2;
    let pie = d3.layout.pie()
        .sort(null)
        .value(d => d.value);
    let data = this.props.data;
    let sectors = pie(data).map((d, i) => {
      return <Sector key={i} data={d} fill={data[i].fill} radius={radius} thickness={this.props.thickness}/>;
    });
    return <svg width={this.props.size} height={this.props.size}>
      <g transform={`translate(${radius}, ${radius})`}>{sectors}</g>
    </svg>;
  }
});
