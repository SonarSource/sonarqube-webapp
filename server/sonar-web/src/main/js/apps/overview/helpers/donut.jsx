import React from 'react';

const Sector = React.createClass({
  render() {
    const arc = d3.svg.arc()
        .outerRadius(this.props.radius)
        .innerRadius(this.props.radius - this.props.thickness);
    return <path d={arc(this.props.data)} style={{ fill: this.props.fill }}></path>;
  }
});

export default React.createClass({
  getDefaultProps() {
    return {
      size: 30,
      thickness: 6
    };
  },

  render() {
    const radius = this.props.size / 2;
    const pie = d3.layout.pie().sort(null)
        .value(d => {
          return d.value
        });
    const data = this.props.data;
    const sectors = pie(data).map((d, i) => {
      return <Sector
          key={i}
          data={d}
          fill={data[i].fill}
          radius={radius}
          thickness={this.props.thickness}/>;
    });
    return (
        <svg width={this.props.size} height={this.props.size}>
          <g transform={`translate(${radius}, ${radius})`}>{sectors}</g>
        </svg>
    );
  }
});
