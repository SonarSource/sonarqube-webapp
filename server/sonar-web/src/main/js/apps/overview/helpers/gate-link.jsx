import React from 'react';

export default React.createClass({
  render() {
    const url = `${baseUrl}/quality_gates/show/${this.props.gate}`;
    return <a href={url}>{this.props.children}</a>;
  }
});
