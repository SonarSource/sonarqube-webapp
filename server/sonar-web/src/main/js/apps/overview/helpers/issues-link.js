import React from 'react';

export default React.createClass({
  render() {
    let params = Object.keys(this.props.params).map((key) => {
          return `${key}=${encodeURIComponent(this.props.params[key])}`;
        }).join('|'),
        url = `${baseUrl}/component_issues/index?id=${encodeURIComponent(this.props.component)}#${params}`;
    return <a href={url}>{this.props.children}</a>;
  }
});
