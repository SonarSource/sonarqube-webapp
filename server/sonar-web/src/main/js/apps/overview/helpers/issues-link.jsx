import React from 'react';

export default React.createClass({
  render: function () {
    var params = Object.keys(this.props.params).map((key) => {
      return `${key}=${encodeURIComponent(this.props.params[key])}`;
    }).join('|');
    var url = `${baseUrl}/component_issues/index?id=${encodeURIComponent(this.props.component)}#${params}`;
    return <a href={url}>{this.props.children}</a>;
  }
});
