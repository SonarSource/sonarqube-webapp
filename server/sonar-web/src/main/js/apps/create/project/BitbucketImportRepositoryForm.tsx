/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as classNames from 'classnames';
import { uniq, without } from 'lodash';
import * as React from 'react';
import { Link } from 'react-router';
import BoxedGroupAccordion from 'sonar-ui-common/components/controls/BoxedGroupAccordion';
import { ButtonLink } from 'sonar-ui-common/components/controls/buttons';
import Radio from 'sonar-ui-common/components/controls/Radio';
import CheckIcon from 'sonar-ui-common/components/icons/CheckIcon';
import { Alert } from 'sonar-ui-common/components/ui/Alert';
import { translate } from 'sonar-ui-common/helpers/l10n';
import { colors } from '../../../app/theme';
import { getProjectUrl } from '../../../helpers/urls';
import { BitbucketProject, BitbucketRepository } from '../../../types/alm-integration';

export interface BitbucketImportRepositoryFormProps {
  importing?: boolean;
  onSelectRepository: (repo: BitbucketRepository) => void;
  projects?: BitbucketProject[];
  projectRepositories?: T.Dict<BitbucketRepository[]>;
  selectedRepository?: BitbucketRepository;
}

export default function BitbucketImportRepositoryForm(props: BitbucketImportRepositoryFormProps) {
  const { importing, projects = [], projectRepositories = {}, selectedRepository } = props;
  const [openProjectKeys, setOpenProjectKeys] = React.useState(
    projects.length > 0 ? [projects[0].key] : []
  );

  if (projects.length === 0) {
    return (
      <Alert variant="warning">{translate('onboarding.create_project.no_bbs_projects')}</Alert>
    );
  }

  const allAreExpanded = projects.length === openProjectKeys.length;

  return (
    <div className="create-project-import-bbs">
      <div className="overflow-hidden spacer-bottom">
        <ButtonLink
          className="pull-right"
          onClick={() => setOpenProjectKeys(allAreExpanded ? [] : projects.map(p => p.key))}>
          {allAreExpanded ? translate('collapse_all') : translate('expand_all')}
        </ButtonLink>
      </div>

      {projects.map(project => {
        const isOpen = openProjectKeys.includes(project.key);
        const repositories = projectRepositories[project.key] || [];

        return (
          <BoxedGroupAccordion
            className={classNames({ open: isOpen })}
            key={project.key}
            onClick={() =>
              setOpenProjectKeys(
                isOpen
                  ? without(openProjectKeys, project.key)
                  : uniq([...openProjectKeys, project.key])
              )
            }
            open={isOpen}
            title={<h3>{project.name}</h3>}>
            {isOpen && (
              <div className="display-flex-wrap">
                {repositories.length === 0 && (
                  <Alert variant="warning">
                    {translate('onboarding.create_project.no_bbs_repos')}
                  </Alert>
                )}

                {repositories.map(repo =>
                  repo.sqProjectKey ? (
                    <span
                      className="display-inline-flex-start spacer-right spacer-bottom create-project-import-bbs-repo"
                      key={repo.id}>
                      <CheckIcon className="spacer-right" fill={colors.green} size={14} />
                      <span>
                        <div className="little-spacer-bottom">
                          <strong>
                            <Link to={getProjectUrl(repo.sqProjectKey)}>{repo.name}</Link>
                          </strong>
                        </div>
                        <em>{translate('onboarding.create_project.repository_imported')}</em>
                      </span>
                    </span>
                  ) : (
                    <Radio
                      checked={selectedRepository?.id === repo.id}
                      className={classNames(
                        'display-inline-flex-start spacer-right spacer-bottom create-project-import-bbs-repo overflow-hidden',
                        {
                          disabled: importing,
                          'text-muted': importing,
                          'link-no-underline': importing
                        }
                      )}
                      key={repo.id}
                      onCheck={() => props.onSelectRepository(repo)}
                      value={String(repo.id)}>
                      <strong className="text-ellipsis">{repo.name}</strong>
                    </Radio>
                  )
                )}
              </div>
            )}
          </BoxedGroupAccordion>
        );
      })}
    </div>
  );
}
