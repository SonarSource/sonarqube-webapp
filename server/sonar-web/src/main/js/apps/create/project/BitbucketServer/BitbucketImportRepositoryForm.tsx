/*
 * SonarQube
 * Copyright (C) 2009-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { Link, LinkHighlight } from '@sonarsource/echoes-react';
import { FormattedMessage, useIntl } from 'react-intl';
import { FlagMessage, InputSearch } from '~design-system';
import { queryToSearchString } from '~sonar-aligned/helpers/urls';
import { BitbucketRepository } from '../../../../types/alm-integration';
import { CreateProjectModes } from '../types';
import BitbucketRepositories from './BitbucketRepositories';

export interface BitbucketImportRepositoryFormProps {
  canFetchMore: boolean;
  hasProjects: boolean;
  onFetchMore: () => void;
  onImportRepository: (repo: BitbucketRepository) => void;
  onSearch: (query: '') => void;
  repositories: BitbucketRepository[];
  searching: boolean;
}

export default function BitbucketImportRepositoryForm(
  props: Readonly<BitbucketImportRepositoryFormProps>,
) {
  const {
    canFetchMore,
    hasProjects,
    onFetchMore,
    onImportRepository,
    onSearch,
    repositories,
    searching,
  } = props;

  const { formatMessage } = useIntl();

  if (!hasProjects) {
    return (
      <FlagMessage variant="warning">
        <span>
          <FormattedMessage
            id="onboarding.create_project.no_bbs_projects"
            values={{
              link: (
                <Link
                  highlight={LinkHighlight.Default}
                  to={{
                    pathname: '/projects/create',
                    search: queryToSearchString({
                      mode: CreateProjectModes.BitbucketServer,
                      resetPat: 1,
                    }),
                  }}
                >
                  <FormattedMessage id="onboarding.create_project.update_your_token" />
                </Link>
              ),
            }}
          />
        </span>
      </FlagMessage>
    );
  }

  return (
    <div>
      <div className="sw-mb-10 sw-w-abs-800 sw-flex">
        <InputSearch
          onChange={onSearch}
          placeholder={formatMessage({
            id: 'onboarding.create_project.search_repositories_by_name',
          })}
          searchInputAriaLabel={formatMessage({
            id: 'onboarding.create_project.search_repositories_by_name',
          })}
          size="full"
        />
      </div>

      <BitbucketRepositories
        canFetchMore={canFetchMore}
        isLoading={searching}
        onFetchMore={onFetchMore}
        onImportRepository={onImportRepository}
        repositories={repositories}
      />
    </div>
  );
}
