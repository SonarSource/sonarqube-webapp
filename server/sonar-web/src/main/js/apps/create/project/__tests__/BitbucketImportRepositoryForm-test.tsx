/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
/* eslint-disable sonarjs/no-duplicate-string */
import { shallow } from 'enzyme';
import * as React from 'react';
import BoxedGroupAccordion from 'sonar-ui-common/components/controls/BoxedGroupAccordion';
import Radio from 'sonar-ui-common/components/controls/Radio';
import { click } from 'sonar-ui-common/helpers/testUtils';
import {
  mockBitbucketProject,
  mockBitbucketRepository
} from '../../../../helpers/mocks/alm-integrations';
import BitbucketImportRepositoryForm, {
  BitbucketImportRepositoryFormProps
} from '../BitbucketImportRepositoryForm';

it('should render correctly', () => {
  expect(shallowRender()).toMatchSnapshot('default');
  expect(shallowRender({ importing: true })).toMatchSnapshot('importing');
  expect(shallowRender({ projects: [] })).toMatchSnapshot('no projects');
  expect(shallowRender({ projectRepositories: {} })).toMatchSnapshot('no repos');
  expect(shallowRender({ selectedRepository: mockBitbucketRepository() })).toMatchSnapshot(
    'selected repo'
  );
});

it('should correctly handle opening/closing accordions', () => {
  const wrapper = shallowRender();
  click(wrapper.find(BoxedGroupAccordion).at(1));
  expect(wrapper).toMatchSnapshot('2nd opened');
});

it('should correctly handle selecting repos', () => {
  const onSelectRepository = jest.fn();
  const repo = mockBitbucketRepository();
  const wrapper = shallowRender({
    onSelectRepository,
    projectRepositories: {
      project: [repo]
    }
  });

  wrapper
    .find(Radio)
    .at(0)
    .prop<Function>('onCheck')();
  expect(onSelectRepository).toBeCalledWith(repo);
});

function shallowRender(props: Partial<BitbucketImportRepositoryFormProps> = {}) {
  return shallow<BitbucketImportRepositoryFormProps>(
    <BitbucketImportRepositoryForm
      onSelectRepository={jest.fn()}
      projectRepositories={{
        project: [
          mockBitbucketRepository(),
          mockBitbucketRepository({ id: 2, slug: 'bar', name: 'Bar', sqProjectKey: 'bar' })
        ]
      }}
      projects={[
        mockBitbucketProject(),
        mockBitbucketProject({ id: 2, key: 'project2', name: 'Project 2' })
      ]}
      {...props}
    />
  );
}
