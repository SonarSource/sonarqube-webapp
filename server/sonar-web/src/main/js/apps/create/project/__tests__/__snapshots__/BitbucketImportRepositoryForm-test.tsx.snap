// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly handle opening/closing accordions: 2nd opened 1`] = `
<div
  className="create-project-import-bbs"
>
  <div
    className="overflow-hidden spacer-bottom"
  >
    <ButtonLink
      className="pull-right"
      onClick={[Function]}
    >
      collapse_all
    </ButtonLink>
  </div>
  <BoxedGroupAccordion
    className="open"
    key="project"
    onClick={[Function]}
    open={true}
    title={
      <h3>
        Project
      </h3>
    }
  >
    <div
      className="display-flex-wrap"
    >
      <Radio
        checked={false}
        className="display-inline-flex-start spacer-right spacer-bottom create-project-import-bbs-repo overflow-hidden"
        key="1"
        onCheck={[Function]}
        value="1"
      >
        <strong
          className="text-ellipsis"
        >
          Repo
        </strong>
      </Radio>
      <span
        className="display-inline-flex-start spacer-right spacer-bottom create-project-import-bbs-repo"
        key="2"
      >
        <CheckIcon
          className="spacer-right"
          fill="#00aa00"
          size={14}
        />
        <span>
          <div
            className="little-spacer-bottom"
          >
            <strong>
              <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to={
                  Object {
                    "pathname": "/dashboard",
                    "query": Object {
                      "branch": undefined,
                      "id": "bar",
                    },
                  }
                }
              >
                Bar
              </Link>
            </strong>
          </div>
          <em>
            onboarding.create_project.repository_imported
          </em>
        </span>
      </span>
    </div>
  </BoxedGroupAccordion>
  <BoxedGroupAccordion
    className="open"
    key="project2"
    onClick={[Function]}
    open={true}
    title={
      <h3>
        Project 2
      </h3>
    }
  >
    <div
      className="display-flex-wrap"
    >
      <Alert
        variant="warning"
      >
        onboarding.create_project.no_bbs_repos
      </Alert>
    </div>
  </BoxedGroupAccordion>
</div>
`;

exports[`should render correctly: default 1`] = `
<div
  className="create-project-import-bbs"
>
  <div
    className="overflow-hidden spacer-bottom"
  >
    <ButtonLink
      className="pull-right"
      onClick={[Function]}
    >
      expand_all
    </ButtonLink>
  </div>
  <BoxedGroupAccordion
    className="open"
    key="project"
    onClick={[Function]}
    open={true}
    title={
      <h3>
        Project
      </h3>
    }
  >
    <div
      className="display-flex-wrap"
    >
      <Radio
        checked={false}
        className="display-inline-flex-start spacer-right spacer-bottom create-project-import-bbs-repo overflow-hidden"
        key="1"
        onCheck={[Function]}
        value="1"
      >
        <strong
          className="text-ellipsis"
        >
          Repo
        </strong>
      </Radio>
      <span
        className="display-inline-flex-start spacer-right spacer-bottom create-project-import-bbs-repo"
        key="2"
      >
        <CheckIcon
          className="spacer-right"
          fill="#00aa00"
          size={14}
        />
        <span>
          <div
            className="little-spacer-bottom"
          >
            <strong>
              <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to={
                  Object {
                    "pathname": "/dashboard",
                    "query": Object {
                      "branch": undefined,
                      "id": "bar",
                    },
                  }
                }
              >
                Bar
              </Link>
            </strong>
          </div>
          <em>
            onboarding.create_project.repository_imported
          </em>
        </span>
      </span>
    </div>
  </BoxedGroupAccordion>
  <BoxedGroupAccordion
    className=""
    key="project2"
    onClick={[Function]}
    open={false}
    title={
      <h3>
        Project 2
      </h3>
    }
  />
</div>
`;

exports[`should render correctly: importing 1`] = `
<div
  className="create-project-import-bbs"
>
  <div
    className="overflow-hidden spacer-bottom"
  >
    <ButtonLink
      className="pull-right"
      onClick={[Function]}
    >
      expand_all
    </ButtonLink>
  </div>
  <BoxedGroupAccordion
    className="open"
    key="project"
    onClick={[Function]}
    open={true}
    title={
      <h3>
        Project
      </h3>
    }
  >
    <div
      className="display-flex-wrap"
    >
      <Radio
        checked={false}
        className="display-inline-flex-start spacer-right spacer-bottom create-project-import-bbs-repo overflow-hidden disabled text-muted link-no-underline"
        key="1"
        onCheck={[Function]}
        value="1"
      >
        <strong
          className="text-ellipsis"
        >
          Repo
        </strong>
      </Radio>
      <span
        className="display-inline-flex-start spacer-right spacer-bottom create-project-import-bbs-repo"
        key="2"
      >
        <CheckIcon
          className="spacer-right"
          fill="#00aa00"
          size={14}
        />
        <span>
          <div
            className="little-spacer-bottom"
          >
            <strong>
              <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to={
                  Object {
                    "pathname": "/dashboard",
                    "query": Object {
                      "branch": undefined,
                      "id": "bar",
                    },
                  }
                }
              >
                Bar
              </Link>
            </strong>
          </div>
          <em>
            onboarding.create_project.repository_imported
          </em>
        </span>
      </span>
    </div>
  </BoxedGroupAccordion>
  <BoxedGroupAccordion
    className=""
    key="project2"
    onClick={[Function]}
    open={false}
    title={
      <h3>
        Project 2
      </h3>
    }
  />
</div>
`;

exports[`should render correctly: no projects 1`] = `
<Alert
  variant="warning"
>
  onboarding.create_project.no_bbs_projects
</Alert>
`;

exports[`should render correctly: no repos 1`] = `
<div
  className="create-project-import-bbs"
>
  <div
    className="overflow-hidden spacer-bottom"
  >
    <ButtonLink
      className="pull-right"
      onClick={[Function]}
    >
      expand_all
    </ButtonLink>
  </div>
  <BoxedGroupAccordion
    className="open"
    key="project"
    onClick={[Function]}
    open={true}
    title={
      <h3>
        Project
      </h3>
    }
  >
    <div
      className="display-flex-wrap"
    >
      <Alert
        variant="warning"
      >
        onboarding.create_project.no_bbs_repos
      </Alert>
    </div>
  </BoxedGroupAccordion>
  <BoxedGroupAccordion
    className=""
    key="project2"
    onClick={[Function]}
    open={false}
    title={
      <h3>
        Project 2
      </h3>
    }
  />
</div>
`;

exports[`should render correctly: selected repo 1`] = `
<div
  className="create-project-import-bbs"
>
  <div
    className="overflow-hidden spacer-bottom"
  >
    <ButtonLink
      className="pull-right"
      onClick={[Function]}
    >
      expand_all
    </ButtonLink>
  </div>
  <BoxedGroupAccordion
    className="open"
    key="project"
    onClick={[Function]}
    open={true}
    title={
      <h3>
        Project
      </h3>
    }
  >
    <div
      className="display-flex-wrap"
    >
      <Radio
        checked={true}
        className="display-inline-flex-start spacer-right spacer-bottom create-project-import-bbs-repo overflow-hidden"
        key="1"
        onCheck={[Function]}
        value="1"
      >
        <strong
          className="text-ellipsis"
        >
          Repo
        </strong>
      </Radio>
      <span
        className="display-inline-flex-start spacer-right spacer-bottom create-project-import-bbs-repo"
        key="2"
      >
        <CheckIcon
          className="spacer-right"
          fill="#00aa00"
          size={14}
        />
        <span>
          <div
            className="little-spacer-bottom"
          >
            <strong>
              <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to={
                  Object {
                    "pathname": "/dashboard",
                    "query": Object {
                      "branch": undefined,
                      "id": "bar",
                    },
                  }
                }
              >
                Bar
              </Link>
            </strong>
          </div>
          <em>
            onboarding.create_project.repository_imported
          </em>
        </span>
      </span>
    </div>
  </BoxedGroupAccordion>
  <BoxedGroupAccordion
    className=""
    key="project2"
    onClick={[Function]}
    open={false}
    title={
      <h3>
        Project 2
      </h3>
    }
  />
</div>
`;
